java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcToArrowArrayTest.java

```diff
@@ -0,0 +1,372 @@
+  }
+
+  @Test
+  public void testJdbcToArrow() throws Exception {
+    int rowCount = 4;
+
+    Integer[][] intArrays = generateIntegerArrayField(rowCount);
+    Float[][] floatArrays = generateFloatArrayField(rowCount);
+    String[][] strArrays = generateStringArrayField(rowCount);
+
+    insertRows(rowCount, intArrays, floatArrays, strArrays);
+
+    final JdbcToArrowConfigBuilder builder =
+        new JdbcToArrowConfigBuilder(new RootAllocator(Integer.MAX_VALUE), JdbcToArrowUtils.getUtcCalendar(), false);
+    builder.setArraySubTypeByColumnNameMap(arrayFieldMapping);
+
+    final JdbcToArrowConfig config = builder.build();
+
+    try (ResultSet resultSet = conn.createStatement().executeQuery(QUERY)) {
+      final VectorSchemaRoot vector;
+
+      try {
+        vector = JdbcToArrow.sqlToArrow(resultSet, config);
+      } catch (Exception e) {
+        e.printStackTrace();
+        throw e;
+      }
+
+      assertEquals(rowCount, vector.getRowCount());
+
+      testIntegerList((ListVector) vector.getVector(INT_ARRAY_FIELD_NAME), rowCount, intArrays);
+      testFloatList((ListVector) vector.getVector(FLOAT_ARRAY_FIELD_NAME), rowCount, floatArrays);
+      testStringList((ListVector) vector.getVector(STRING_ARRAY_FIELD_NAME), rowCount, strArrays);
+    }
+  }
+
+  @Test
+  public void testJdbcToArrowWithNulls() throws Exception {
+    int rowCount = 3;
+    Integer[][] intArrays = new Integer[rowCount][];
+    intArrays[0] = null;
+    intArrays[1] = new Integer[] { 0 };
+    intArrays[2] = new Integer[] { 1 };
+
+    Float[][] floatArrays = new Float[rowCount][];
+    floatArrays[0] = new Float[] { 2.0f };
+    floatArrays[1] = null;
+    floatArrays[2] = new Float[] { 3.0f };
+
+    String[][] stringArrays = new String[rowCount][];
+    stringArrays[0] = new String[] { "4" };
+    stringArrays[1] = null;
+    stringArrays[2] = new String[] { "5" };
+
+    insertRows(rowCount, intArrays, floatArrays, stringArrays);
+
+    final JdbcToArrowConfigBuilder builder =
+        new JdbcToArrowConfigBuilder(new RootAllocator(Integer.MAX_VALUE), JdbcToArrowUtils.getUtcCalendar(), false);
+    builder.setArraySubTypeByColumnNameMap(arrayFieldMapping);
+
+    final JdbcToArrowConfig config = builder.build();
+
+    try (ResultSet resultSet = conn.createStatement().executeQuery(QUERY)) {
+      final VectorSchemaRoot vector;
+
+      try {
+        vector = JdbcToArrow.sqlToArrow(resultSet, config);
+      } catch (Exception e) {
+        e.printStackTrace();
+        throw e;
+      }
+
+      assertEquals(rowCount, vector.getRowCount());
+
+      testIntegerList((ListVector) vector.getVector(INT_ARRAY_FIELD_NAME), rowCount, intArrays);
+      testFloatList((ListVector) vector.getVector(FLOAT_ARRAY_FIELD_NAME), rowCount, floatArrays);
+      testStringList((ListVector) vector.getVector(STRING_ARRAY_FIELD_NAME), rowCount, stringArrays);
+    }
+  }
+
+  private void testIntegerList(ListVector listVector, int rowCount, Integer[][] expectedValues) {
```

Comment:
assertIntegerVectorEquals might be a clearer name for this method.

Reply:
Changed.